### PR TITLE
feat: Make the `to_json` filter support the conf object

### PR DESCRIPTION
**Pull request by pawamoy**
Fixes issue #389

Any idea for a test :smile:? I guess I'll just run the filter in a template.

### CODE REVIEW

File: copier/main.py [hunk 1]
```diff
@@ -400,6 +410,11 @@ def jinja_env(self) -> SandboxedEnvironment:
                 "Make sure to install these extensions alongside Copier itself.\n"
                 "See the docs at https://copier.readthedocs.io/en/latest/configuring/#jinja_extensions"
             )
+        # patch the `to_json` filter to support Pydantic dataclasses
+        def to_json(value, *args, **kwargs):
+            return json.dumps(value, *args, **kwargs, default=pydantic_encoder)
+
+        env.filters["to_json"] = to_json
```
Comment (Yajo) [inline copier/main.py:417]: I think you can simplify it with a partial:

```suggestion
        env.filters["to_json"] = partial(env.filters["to_json"], default=pydantic_encoder)
```

**Review by Yajo — COMMENTED** [review]
It looks awesome! Thanks, this was one of the blockers for v6.

The test could be simple: just create a `whatever.json.jinja` file with `{{ _copier_conf | to_json }}" inside and check it dumps a valid json object.

File: copier/main.py (hunk 1 above)
Comment (pawamoy) [reply copier/main.py:417]: Nice, thanks!

**Comment by pawamoy** [comment]
I'm afraid keeping the `.json` method for backward-compatibility is problematic: the pydantic encoder then fails because it cannot serialize the `.json` method on the conf object. Should we simply drop it? Otherwise we must derive the pydantic encoder to ignore functions (or copy the conf object and remove the `.json` attribute from the copy).

**Review by Yajo — APPROVED** [review]
It's not my favorite solution because it can introduce big incompatibilities with v5, but after all it's a major release, so nevermind. If it's a real problem, somebody can fix it later.

BTW this is still draft. Anything left behind?

**Comment by pawamoy** [comment]
It's ready now.

**Pull request merged**